action/protocol/execution/evm/evm.go

```diff
@@ -418,6 +418,12 @@ func executeInEVM(ctx context.Context, evmParams *Params, stateDB *StateDBAdapte
 		config = vmCfg
 	}
 	chainConfig := getChainConfig(g, blockHeight, evmParams.evmNetworkID)
+	if g.IsRedsea(blockHeight) {
+		// Merge enabled at Redsea height
+		tipHash := protocol.MustGetBlockchainCtx(ctx).Tip.Hash
+		h := common.BytesToHash(tipHash[:])
+		evmParams.context.Random = &h
```

Comment:
The initialization of the `Random` field should occur in where `evmParams` is created.

Reply:
moved to `newParam`